util/pkg/vfs/context.go

```diff
@@ -95,32 +98,93 @@ func (c *VFSContext) BuildVfsPath(p string) (Path, error) {
 
 // readHttpLocation reads an http (or https) url.
 // It returns the contents, or an error on any non-200 response.  On a 404, it will return os.ErrNotExist
+// It will retry a few times on a 500 class error
 func (c *VFSContext) readHttpLocation(httpURL string, httpHeaders map[string]string) ([]byte, error) {
-	req, err := http.NewRequest("GET", httpURL, nil)
-	if err != nil {
-		return nil, err
-	}
-	for k, v := range httpHeaders {
-		req.Header.Add(k, v)
+	// Exponential backoff, starting with 500 milliseconds, doubling each time, 5 steps
+	backoff := wait.Backoff{
+		Duration: 500 * time.Millisecond,
+		Factor:   2,
+		Steps:    5,
 	}
-	response, err := http.DefaultClient.Do(req)
-	if response != nil {
-		defer response.Body.Close()
-	}
-	if err != nil {
-		return nil, fmt.Errorf("error fetching %q: %v", httpURL, err)
-	}
-	body, err := ioutil.ReadAll(response.Body)
+
+	var body []byte
+
+	done, err := RetryWithBackoff(backoff, func() (bool, error) {
+		glog.V(4).Infof("Performing HTTP request: GET %s", httpURL)
+		req, err := http.NewRequest("GET", httpURL, nil)
+		if err != nil {
+			return false, err
+		}
+		for k, v := range httpHeaders {
+			req.Header.Add(k, v)
+		}
+		response, err := http.DefaultClient.Do(req)
+		if response != nil {
+			defer response.Body.Close()
+		}
+		if err != nil {
+			return false, fmt.Errorf("error fetching %q: %v", httpURL, err)
+		}
+		body, err = ioutil.ReadAll(response.Body)
+		if err != nil {
+			return false, fmt.Errorf("error reading response for %q: %v", httpURL, err)
+		}
+		if response.StatusCode == 404 {
+			// We retry on 404s in case of eventual consistency
+			return true, os.ErrNotExist
+		}
+		if response.StatusCode >= 500 && response.StatusCode <= 599 {
+			// Retry on 5XX errors
+			return false, fmt.Errorf("unexpected response code %q for %q: %v", response.Status, httpURL, string(body))
+		}
+
+		if response.StatusCode == 200 {
+			return true, nil
+		}
+
+		// Don't retry on other errors
+		return true, fmt.Errorf("unexpected response code %q for %q: %v", response.Status, httpURL, string(body))
+	})
 	if err != nil {
-		return nil, fmt.Errorf("error reading response for %q: %v", httpURL, err)
-	}
-	if response.StatusCode == 404 {
-		return nil, os.ErrNotExist
+		return nil, err
+	} else if done {
+		return body, nil
+	} else {
+		// Shouldn't happen - we always return a non-nil error with false
+		return nil, wait.ErrWaitTimeout
 	}
-	if response.StatusCode != 200 {
-		return nil, fmt.Errorf("unexpected response code %q for %q: %v", response.Status, httpURL, string(body))
+}
+
+// RetryWithBackoff runs until a condition function returns true, or until Steps attempts have been taken
+// As compared to wait.ExponentialBackoff, this function returns the results from the function on the final attempt
+func RetryWithBackoff(backoff wait.Backoff, condition func() (bool, error)) (bool, error) {
```

Comment:
One day we will have standardized polling.. I am surprised this isn't in k8s yet :) 

Code looks solid - thanks Justin